src/main/php/lang/ast/emit/PHP.class.php

```diff
@@ -622,13 +644,18 @@ protected function emitMethod($result, $method) {
       }
       $this->emitAll($result, $method->body);
       $result->out->write('}');
+      $default= null;
     }
 
     foreach ($promoted as $param) {
       $this->emitProperty($result, new Property(explode(' ', $param->promote), $param->name, $param->type));
     }
 
     $result->locals= array_pop($result->stack);
+
+    // Copy default implementations to class scope
+    $default && $result->locals[3][]= $default;
```

Comment:
Can this use a property inside `lang.ast.emit.InType` instead of misusing locals?